lib/puppet/provider/service/systemd.rb

```diff
@@ -28,12 +28,33 @@ def disable
 
   def enabled?
     begin
-      systemctl("is-enabled", @resource[:name])
+      systemctl_info = systemctl(
+         'show',
+         @resource[:name],
+         '--property', 'LoadState',
+         '--property', 'UnitFileState',
+         '--no-pager'
+      )
+
+      svc_info = Hash.new
+      systemctl_info.split.map{|svc|
```

Comment:
`#map` is used to transform one enumerable object to another, but in this case we're ignoring the return value and modifying `svc_info` instead. Would `#each` be more appropriate?